miniwallet/test/Mini721.ts

```diff
@@ -0,0 +1,971 @@
+      releaseCount = 5
+      await communityMint721(this, this.dev.address, releaseCount)
+      await this.mini721.setOffsetValue(2200)
+      // set state variables
+      await this.mini721.toggleSaleState()
+      // Set the random start Index
+      await this.mini721.setTemporaryTokenUri(
+        'ipfs://TemporaryTokenUriPlaceHolder/'
+      )
+      await this.mini721.setStartIndex()
+      expect(await this.mini721.saleIsActive()).to.equal(true)
+      expect(await this.mini721.totalSupply()).to.equal(2200)
+      expect(await this.mini721.exists(0)).to.equal(true)
+      // mint tokens
+      releaseCount = 1
+      await userMint721(this, this.alice, releaseCount)
+
+      expect(await this.mini721.totalSupply()).to.equal(2201)
+      releaseCount = 5
+      await userMint721(this, this.bob, releaseCount)
+      // const mintPrice = await this.mini721.mintPrice();
+      // await this.mini721.connect(this.bob).mintMini(releaseCount, {
+      //   value: mintPrice.mul(releaseCount),
+      // });
+
+      expect(await this.mini721.totalSupply()).to.equal(2206)
+      releaseCount = 10
+      await userMint721(this, this.carol, releaseCount)
+
+      releaseCount = 1
+      expect(await this.mini721.totalSupply()).to.equal(2216)
+      expect(await this.mini721.maxMiniTokens()).to.equal(10000)
+      await userMint721(this, this.alice, releaseCount)
+
+      // turn sale off and check user can no longer mint
+      await this.mini721.toggleSaleState()
+      expect(await this.mini721.saleIsActive()).to.equal(false)
+      await expect(
+        this.mini721.connect(this.alice).mintMini(releaseCount, {
+          value: ethers.utils.parseEther('0.1').mul(releaseCount)
+        })
+      ).to.be.revertedWith('Mini721: Sale is not active')
+
+      // Public Variables
+      expect(await this.mini721.maxMiniTokens()).to.equal(10000)
+      expect(await this.mini721.mintPrice()).to.equal(
+        ethers.utils.parseEther('0.1')
+      ) // 0.1 ETH
+      expect(await this.mini721.maxPerMint()).to.equal(10)
+      expect(await this.mini721.provenanceHash()).to.equal('')
+      expect(await this.mini721.startIndex()).to.not.equal(0)
+      console.log(`startIndex: ${await this.mini721.startIndex()}`)
+      expect(await this.mini721.offsetValue()).to.equal(2200)
+      expect(await this.mini721.metadataFrozen()).to.equal(false)
+      expect(await this.mini721.provenanceFrozen()).to.equal(false)
+      expect(await this.mini721.saleIsActive()).to.equal(false)
+      expect(await this.mini721.saleStarted()).to.equal(true)
+      expect(await this.mini721.temporaryTokenUri()).to.equal(
+        'ipfs://TemporaryTokenUriPlaceHolder/'
+      )
+
+      // Public Functions Mini721
+      expect(await this.mini721.tokenURI(12)).to.equal(
+        'ipfs://TemporaryTokenUriPlaceHolder/'
+      )
+
+      // TODO review supportsInterface logic
+      expect(await this.mini721.supportsInterface('0x2a55205a')).to.equal(true)
```

Comment:
should have a constant mapping somewhere to human-readable methods

Reply:
Added this in constants 🙏